api.js

```diff
@@ -426,6 +427,64 @@ API.prototype.noticeError = function noticeError(error, customAttributes) {
   this.agent.errors.addUserError(transaction, error, filteredAttributes)
 }
 
+/**
+ * Sends an application log message to New Relic. The agent already
+ * automatically does this for some instrumented logging libraries,
+ * but in case you are using another logging method that is not
+ * already instrumented by the agent, you can use this function
+ * instead.
+ *
+ * If application log forwarding is disabled in the agent
+ * configuration, this function does nothing.
+ *
+ * @param {object} logMessage  The log line object to send.
+ * @param {string} logMessage.message The log message.
+ * @param {string} logMessage.level The log level severity. If this key is
+ *   missing, it will default to UNKNOWN
+ * @param {number} logMessage.timestamp   Unix epoch number denoting the
+ *   time that this log message was produced. If this key is missing,
+ *   it will default to the output of `Date.now()`.
+ */
+API.prototype.recordLogEvent = function recordLogEvent(logMessage) {
+  const metric = this.agent.metrics.getOrCreateMetric(NAMES.SUPPORTABILITY.API + '/recordLogEvent')
+  metric.incrementCallCount()
+
+  if (!applicationLogging.isLogForwardingEnabled(this.agent.config, this.agent)) {
+    logger.warnOnce(
+      'Record logs',
+      'Application log forwarding disabled, method API#recordLogEvent will not record messages'
+    )
+    return
+  }
+
+  // If they don't pass a logMessage object, or it doesn't have the
+  // required `message` key, bail out.
+  if (!logMessage || typeof logMessage !== 'object' || logMessage.message === undefined) {
+    logger.warn(
+      'recordLogEvent requires an object with a `message` attribute for its single argument, got %s (%s)',
+      stringify(logMessage),
+      typeof logMessage
+    )
+    return
+  }
+
+  if (logMessage.level === undefined) {
```

Comment:
level must be a certain value and if it's not it should be marked as unknown. it may get messy as I think different logging libraries have all different levels.  Currently we technically only support the lower case versions of these [metrics](https://github.com/newrelic/node-newrelic/blob/main/lib/metrics/names.js#L276)

Reply:
No change needed here. While those are the only metric names allowed, the spec says, "note these levels are not standard between frameworks so retain the local naming". The logic for incrementing only the named metrics is already handled in `incrementLoggingMetrics`.